peercrawler.py

```diff
@@ -280,7 +280,7 @@ def crawl(self, forever, delay, max_workers=4):
             count += 1
 
     # noinspection PyUnresolvedReferences
-    def get_dot_string(self, only_voting: bool = False) -> str:
+    def get_dot_string(self, filter_function: Callable[[Peer], bool] = None) -> str:
```

Comment:
The filter function is a nice touch!

Reply:
I wonder if the filter function should take both the node and peer or even peers together.
But I am happy to take it one step at the time and not aim too high too early.